### ISSUE TITLE
Crash when clicking permissions in certain websites

### ISSUE THREAD

**Issue by PurpleCodingWizard** [bromite/bromite#1355]
### Bromite version

Version: `92.0.4515.134`
Arch: `arm`
Android version: `10`
Device model: `Redmi 7a`

### Flags

These are the flags changed under `chrome://flags`:
1. Chrome-sharing-hub: Disabled
2. Omibox-switch-to-tab-suggestions: Enabled
3. Reading-list: Enabled
4. Tab-group-UI-improvements: Enabled

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No
### Is this bug happening in an incognito tab?
Yes

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?

Yes

### Describe the bug

Basically clicking the permissions settings on certain websites cause the app to hang and crash to homescreen

### Steps to reproduce the bug

Steps to reproduce the bug:
1. Go to forum.palemoon.com or msn.com
2. Click on the lock icon
3. click on permissions settings

### Expected behavior

Bromite shows the pop-up to toggle settings like JavaScript, block ads, etc

### Actual behaviour

The app hangs and exits to home screen after a while

**Comment by csagan5** [comment]
> * Go to forum.palemoon.com or msn.com
> * Click on the lock icon
> * click on permissions settings

There are no permission settings after clicking the lock icon. How do you reproduce this crash?

**Comment by PurpleCodingWizard** [comment]
I have javascript globally disabled. When I click the lock icon and later click "Permissions" in the pop-up menu it crashes. I include a video to demonstrate the crash here https://imgur.com/a/4CzIkMv @csagan5

**Comment by csagan5** [comment]
Still cannot be reproduced; can you please submit the crash dump? There is a new functionality to get them via `chrome://crashes`

**Comment by PurpleCodingWizard** [comment]
Ok. thanks for the info. Here's the most recently one which was saved uploaded to dropbox: [Bug-report-zipfile](https://www.dropbox.com/s/z5e0156dmz37x0z/f6ea20ea-f410-42a8-89e8-5260ef16a845.zip?dl=0)

**Comment by PurpleCodingWizard** [comment]
Any updates? @csagan5

**Comment by csagan5** [comment]
@PurpleCodingWizard please attach the file here on GitHub

**Comment by PurpleCodingWizard** [comment]
[
[945cdd23-17db-4858-849f-5c77c3fad258.zip](https://github.com/bromite/bromite/files/7230079/945cdd23-17db-4858-849f-5c77c3fad258.zip)
](url) here, hope it helps@csagan5

**Comment by csagan5** [comment]
@uazo the crash seems to be:
```
jni_android.cc:306: Pl
ease include Java exception stack in crash report
exception_info android.content.res.Resources$NotFoundException: String resource ID #0x0
        at android.content.res.Resources.getText(Resources.java:381)
        at android.content.res.MiuiResources.getText(MiuiResources.java:97)
        at android.content.res.Resources.getString(Resources.java:474)
        at c80.D(chromium-ChromePublic.apk-stable-1631381150:1)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.Z0(chromium-ChromePublic.apk-stable-1631381150:9)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.J0(chromium-ChromePublic.apk-stable-1631381150:62)
        at ci1.b(chromium-ChromePublic.apk-stable-1631381150:5)
```

Could it be due to the timezone or user agent settings?

**Comment by uazo** [comment]
yoy can deobfuscate with 
`build/android/stacktrace/java_deobfuscate.py ~/whereis/ChromePublic.apk.mapping <stacktrace.file`

```
android.content.res.Resources$NotFoundException: String resource ID #0x0
        at android.content.res.Resources.getText(Resources.java:381)
        at android.content.res.MiuiResources.getText(MiuiResources.java:97)
        at android.content.res.Resources.getString(Resources.java:474)
        at androidx.fragment.app.Fragment.getString(Fragment.java:1004)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.setupContentSettingsPreference(SingleWebsiteSettings.java:917)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.setUpCookiesPreference(SingleWebsiteSettings.java:1097)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.setupContentSettingsPreferences(SingleWebsiteSettings.java:506)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.displaySitePermissions(SingleWebsiteSettings.java:463)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings.access$100(SingleWebsiteSettings.java:48)
        at org.chromium.components.browser_ui.site_settings.SingleWebsiteSettings$SingleWebsitePermissionsPopulator.onWebsitePermissionsAvailable(SingleWebsiteSettings.java:231)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$PermissionsAvailableCallbackRunner.run(WebsitePermissionsFetcher.java:414)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$Task.runAsync(WebsitePermissionsFetcher.java:295)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$TaskQueue.next(WebsitePermissionsFetcher.java:306)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$Task.runAsync(WebsitePermissionsFetcher.java:296)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$TaskQueue.next(WebsitePermissionsFetcher.java:306)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$Task.runAsync(WebsitePermissionsFetcher.java:296)
        at org.chromium.components.browser_ui.site_settings.WebsitePermissionsFetcher$TaskQueue.next(WebsitePermissionsFetcher.java:306)

```

> Could it be due to the timezone or user agent settings?

the code is modified by the timezone patch but it seems to be related to cookies.
I cannot reproduce the problem

**Comment by csagan5** [comment]
I cannot reproduce it either; I wonder what it means. This is the closest answer I could find: https://stackoverflow.com/a/60492909

**Comment by uazo** [comment]
> This is the closest answer I could find: https://stackoverflow.com/a/60492909

very likely

**Comment by csagan5** [comment]
@PurpleCodingWizard is this still reproducible?

**Comment by uazo** [comment]
I'm sorry, I didn't notice it before!
the bug is probably [here](https://source.chromium.org/chromium/chromium/src/+/main:components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/ContentSettingsResources.java;l=440;drc=2cec6d69836ae3a62b54e8b420c9e39377145190;bpv=1;bpt=1
), the case `ContentSettingValues.DEFAULT` is missing.
@csagan5 do I fix it now or are we waiting for the v96?

**Comment by csagan5** [comment]
@uazo did you already include this in #1606?

I suggest to do this on v96 official patches and to cover all cases in the relative patches introduces the permission.

**Comment by uazo** [comment]
> did you already include this in #1606?

No. it doesn't have much to do with that. 
I think it's the fault of the default cookie setting: I have often seen that in the code the default == enabled, while in bromite it is not.

> I suggest to do this on v96 official patches and to cover all cases in the relative patches introduces the permission

Ok.

**Comment by csagan5** [comment]
Ok; feel free to make a fix for it then on top of v96 official patches.

**Comment by csagan5** [comment]
Fixed in `98.0.4758.108`.

**Comment by Nemris** [comment]
Doesn't quite solve my issue, though.
